sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -1179,8 +1174,10 @@ public WriteJmsResult<T> expand(PCollection<T> input) {
               PUBLISH_TO_JMS_STEP_NAME,
               ParDo.of(new JmsIOProducerFn<>(spec, failedMessagesTag))
                   .withOutputTags(messagesTag, TupleTagList.of(failedMessagesTag)));
-      PCollection<T> failedPublishedMessages =
-          failedPublishedMessagesTuple.get(failedMessagesTag).setCoder(input.getCoder());
+      PCollection<JmsError<T>> failedPublishedMessages =
+          failedPublishedMessagesTuple
```

Comment:
are we good to change the returned PCollection by using the new `JmsError`?

Reply:
this is breaking change, maybe at least communicate it in changes file.

Reply:
I can add one option to control this. Thoughts?

Reply:
@Abacn WDYT?